ScoutSuite/providers/aws/services/s3.py

```diff
@@ -137,6 +133,7 @@ def update_iam_permissions(s3_info, bucket_name, iam_entity, allowed_iam_entity,
 
 def update_bucket_permissions(s3_info, iam_info, action, iam_entity, allowed_iam_entity, full_path, policy_type,
                               policy_name):
+    global policy
```

Comment:
Why do we need a `global` variable?

Reply:
To prevent the warning `Local variable 'policy' might be referenced before assignment` thrown by PyCharm.

Reply:
A better way to do this is to actually assign it to an empty dictionary at the beginning of the function. Pushing this now